src/basic/time-util.h

```diff
@@ -111,20 +111,23 @@ usec_t triple_timestamp_by_clock(triple_timestamp *ts, clockid_t clock);
 
 usec_t timespec_load(const struct timespec *ts) _pure_;
 nsec_t timespec_load_nsec(const struct timespec *ts) _pure_;
-struct timespec *timespec_store(struct timespec *ts, usec_t u);
-struct timespec *timespec_store_nsec(struct timespec *ts, nsec_t n);
+struct timespec* timespec_store(struct timespec *ts, usec_t u);
+struct timespec* timespec_store_nsec(struct timespec *ts, nsec_t n);
 
 usec_t timeval_load(const struct timeval *tv) _pure_;
-struct timeval *timeval_store(struct timeval *tv, usec_t u);
+struct timeval* timeval_store(struct timeval *tv, usec_t u);
 
-char *format_timestamp_style(char *buf, size_t l, usec_t t, TimestampStyle style);
-char *format_timestamp_relative(char *buf, size_t l, usec_t t);
-char *format_timespan(char *buf, size_t l, usec_t t, usec_t accuracy);
+char* format_timestamp_style(char *buf, size_t l, usec_t t, TimestampStyle style);
+char* format_timestamp_relative(char *buf, size_t l, usec_t t);
+char* format_timespan(char *buf, size_t l, usec_t t, usec_t accuracy);
 
-static inline char *format_timestamp(char *buf, size_t l, usec_t t) {
+static inline char* format_timestamp(char *buf, size_t l, usec_t t) {
         return format_timestamp_style(buf, l, t, TIMESTAMP_PRETTY);
 }
 
+/* Use only as function argument, never stand-alone! */
+#define FORMAT_TIMESTAMP(t) format_timestamp((char[FORMAT_TIMESTAMP_MAX]){}, FORMAT_TIMESTAMP_MAX, t)
```

Comment:
why not use stand-alone? As I understand C99 compound literals they remain valid in the immediately surrounding block. i.e. as i read it this is safe:

```c
static const char *foo(char p[2]) {
        p[0] = 'x'; p[1] = 0;
}

int main(int argc, char *argv[]) {
        const char *p;

        p = foo((char[2]) {});
        printf("%s\n", p);
        return 0;
}
```

i.e. the compound literal here will remain allocated until main() returns, since that's the immediately surrounding block.

Reply:
But then there is this:

https://stackoverflow.com/questions/34880638/compound-literal-lifetime-and-if-blocks

Reply:
maybe rename this as FORMAT_TIMESTAMP_AS_LITERAL() or so ? or even FORMAT_TIMESTAMP_AS_COMPOUND_LITERAL()?

Just to highlight the lifecycle guarantees C99 gives there